pkg/redispool/redispool.go

```diff
@@ -49,16 +50,30 @@ func init() {
 	}
 }
 
+var schemeMatcher = regexp.MustCompile("^[A-Za-z][A-Za-z0-9\\+\\-\\.]*://")
+
+// redisDialer returns the appropriate Redis dial function given the raw endpoint string.
+// The string can have two formats:
+// 1) If it begins with "redis://", it is assumed to be of the format specified in
+//    https://www.iana.org/assignments/uri-schemes/prov/redis.
+// 2) Otherwise, it is assumed to be of the format $HOSTNAME:$PORT.
+func redisDialer(rawEndpoint string) func() (redis.Conn, error) {
+	return func() (redis.Conn, error) {
+		if schemeMatcher.MatchString(rawEndpoint) { // expect "redis://"
```

Comment:
could we make this much clearer and just test for the scheme prefix?

```suggestion
		if strings.HasPrefix(rawEndpoint, "redis://") {
```

Edit: I see we probably should also support `rediss://`. So maybe check for both? https://github.com/garyburd/redigo/blob/HEAD/redis/conn.go#L248

Alternatively a `host:port` is not allowed to contain a `/`, so we could use that as the litmus test.

Reply:
There is also case-sensitivity to think about. Pushed up a commit to (1) update the docstring to mention verifying the scheme if it s present and (2) check for "/".